projects/solar/src/mppt.c

```diff
@@ -0,0 +1,42 @@
+#include "gpio.h"
+#include "mux.h"
+#include "spv1020_mppt.h"
+
+// will get the address and size of the demultiplexer
+static MuxAddress s_mux_address = {
+  .bit_width = 3,
+  .sel_pins[0] = { .port = GPIO_PORT_B, .pin = 3 },
+  .sel_pins[1] = { .port = GPIO_PORT_B, .pin = 4 },
+  .sel_pins[2] = { .port = GPIO_PORT_B, .pin = 5 },
```

Comment:
Damn, I had no idea that this syntax works. I'd probably prefer these GPIO addresses to be #define'd so that we can move them easily to a config file with all the hardware details, but this is fine.